server/api/src/rsPhyPathReg.cpp

```diff
@@ -65,1328 +65,1310 @@ namespace ix = irods::experimental;
+                continue;
+            }
 
-    status = getDataObjInfo( rsComm, phyPathRegInp, &dataObjInfoHead,
-                             ACCESS_READ_OBJECT, 0 );
-    if ( status < 0 ) {
-        rodsLog( LOG_ERROR, "%s: getDataObjInfo for [%s] failed",
-                 __FUNCTION__, phyPathRegInp->objPath );
-        return status;
-    }
+            subPhyPathRegInp = *phyPathRegInp;
+            snprintf( subPhyPathRegInp.objPath, MAX_NAME_LEN, "%s/%s",
+                      phyPathRegInp->objPath, rodsDirent->d_name );
 
-    status = sortObjInfoForOpen( &dataObjInfoHead, &phyPathRegInp->condInput, 0 );
-    if ( status < 0 ) {
-        // =-=-=-=-=-=-=-
-        // we perhaps did not match the hier string but
-        // we can still continue as we have a good copy
-        // for a read
-        if ( NULL == dataObjInfoHead ) {
-            return status; // JMC cppcheck - nullptr
-        }
-    }
+            if ( matchPathname( ExcludePatterns, rodsDirent->d_name, filePath ) ) {
+                free( rodsDirent );
+                continue;
+            }
 
-    destDataObjInfo = *dataObjInfoHead;
-    rstrcpy( destDataObjInfo.filePath, filePath, MAX_NAME_LEN );
-    rstrcpy( destDataObjInfo.rescName, _resc_name, NAME_LEN );
-    rstrcpy( destDataObjInfo.rescHier, resc_hier, MAX_NAME_LEN );
-    irods::error ret = resc_mgr.hier_to_leaf_id(resc_hier,destDataObjInfo.rescId);
-    if( !ret.ok() ) {
-        irods::log(PASS(ret));
-    }
+            fileStatInp_t fileStatInp;
+            memset( &fileStatInp, 0, sizeof( fileStatInp ) );
 
-    regReplica_t regReplicaInp{};
-    regReplicaInp.srcDataObjInfo = dataObjInfoHead;
-    regReplicaInp.destDataObjInfo = &destDataObjInfo;
-    if ( getValByKey( &phyPathRegInp->condInput, SU_CLIENT_USER_KW ) != NULL ) {
-        addKeyVal( &regReplicaInp.condInput, SU_CLIENT_USER_KW, "" );
-        addKeyVal( &regReplicaInp.condInput, ADMIN_KW, "" );
-    }
-    else if ( getValByKey( &phyPathRegInp->condInput,
-                           ADMIN_KW ) != NULL ) {
-        addKeyVal( &regReplicaInp.condInput, ADMIN_KW, "" );
-    }
-    // Data size can be passed via DATA_SIZE_KW to save a stat later
-    const auto data_size_str = getValByKey(&phyPathRegInp->condInput, DATA_SIZE_KW);
-    if (nullptr != data_size_str) {
-        addKeyVal(&regReplicaInp.condInput, DATA_SIZE_KW, data_size_str);
-    }
-    if (getValByKey(&phyPathRegInp->condInput, REGISTER_AS_INTERMEDIATE_KW)) {
-        addKeyVal(&regReplicaInp.condInput, REGISTER_AS_INTERMEDIATE_KW, "");
-    }
-    status = rsRegReplica( rsComm, &regReplicaInp );
-    clearKeyVal( &regReplicaInp.condInput );
-    freeAllDataObjInfo( dataObjInfoHead );
+            // Issue #3658 - This section removes trailing slashes from the 
+            // directory path in the server when the path is already in the catalog.
+            //
+            // TODO:  This is a localized fix that addresses any trailing slashes
```

Comment:
ew?  we don't have anything smarter?

Reply:
this might be helpful:  `void remove_trailing_path_separators(char* path)`

it's located in `lib/core/include/rodsPath.h`